File: src/main/java/macro303/keschet/core/pieces/Emperor.java
package macro303.keschet.core.pieces;

import macro303.board_game.Colour;
import macro303.keschet.Direction;
import org.jetbrains.annotations.NotNull;

import static macro303.keschet.Direction.*;

/**
 * Created by Macro303 on 2018-02-08.
 */
public class Emperor extends Piece {
	@NotNull
//	private static final String symbol = "\u265A";
	private static final String symbol = "E";
	private static final int maxDistance = 4;
	@NotNull
	private static final Direction[] validDirections = new Direction[]{NORTH, NORTH_EAST, EAST, SOUTH_EAST, SOUTH, SOUTH_WEST, WEST, NORTH_WEST};

	public Emperor(@NotNull Colour colour) {
		super(colour, maxDistance, symbol, validDirections);
	}

	public Emperor() {
		super(maxDistance, symbol, validDirections);
	}

	@Override
	public String toString() {
		return "Emperor{} " + super.toString();
	}
}

File: Base/src/main/kotlin/macro303/base/pieces/Emperor.kt
package macro303.base.pieces

import macro303.base.Colour
import macro303.base.Direction

class Emperor(teamColour: Colour) : Piece(
	teamColour = teamColour,
	maxDistance = 4,
	symbol = "E",
	validDirections = arrayOf(
		Direction.NORTH,
		Direction.NORTH_EAST,
		Direction.EAST,
		Direction.SOUTH_EAST,
		Direction.SOUTH,
		Direction.SOUTH_WEST,
		Direction.WEST,
		Direction.NORTH_WEST
	)
) {

	override fun equals(other: Any?): Boolean {
		if (this === other) return true
		if (other !is Emperor) return false
		if (!super.equals(other)) return false
		return true
	}

	override fun hashCode(): Int {
		return super.hashCode()
	}

	override fun toString(): String {
		return "Emperor() ${super.toString()}"
	}
}